### DISCUSSION TITLE
BUG: PR viewed files not working properly

### DISCUSSION THREAD

**Discussion by sam-woodridge** [community/community#16811] [Pull Requests] [answered]
When reviewing a PR, the viewed files feature seems to be working incorrectly.

For example, when viewing a PR for the first time, I will see the count correctly show that I have viewed 0 files, yet individual files will show up as viewed:

<img width="346" alt="Screen Shot 2022-05-17 at 4 42 29 PM" src="https://user-images.githubusercontent.com/69694651/168924326-e593000f-4aff-4831-bfb1-c17e39b5d840.png">

Unchecking the box on the file that I haven't viewed yet will make the count go negative, and then the checkbox becomes disabled:

<img width="344" alt="Screen Shot 2022-05-17 at 4 42 43 PM" src="https://user-images.githubusercontent.com/69694651/168924363-bd39773f-fb70-4b18-995f-df7e2bb65219.png">

This issue seems to affect existing PRs that I have already viewed some files on as well. Files that have been changed are marked as viewed even when I haven't viewed them yet.

<img width="363" alt="Screen Shot 2022-05-17 at 4 48 29 PM" src="https://user-images.githubusercontent.com/69694651/168924711-1b33612f-c009-4371-ab24-3a25c2555a05.png">

In the screenshot above, note how the "viewed files" count says 6, when more than 6 files are marked as viewed (in total, there are 15 files the UI marks as viewed, despite the count saying only 6 have been viewed).

Could someone take a look to see what's causing this? It is making the PR process quite difficult as the UI is not properly showing us what we've already viewed, in some cases erroneously telling users they have viewed files they have not. This is not good for proper code review processes.

**Comment by bsilver8192** [comment]
Me and a coworker have both experienced this today. It was working fine for both of us yesterday. I have also observed erroneous "Changed since last view" tags on PRs I have never viewed.

**Reply by SpencerKaiser** [reply]
I almost merged several PRs because I thought I'd already reviewed other files when in reality I'd never even reviewed them at all 🤦‍♂️ 

A dev on my team also saw this lovely gem: 
![image](https://user-images.githubusercontent.com/6445731/168968085-e671fbdc-02ef-43a5-8f9e-1ca8186b13f4.png)

This bug is badddddd... I hope they get it resolved ASAP because it's really negatively impacting my code reviews 😢

**Comment by varun-a-mhatre** [comment]
Got the same issue for one of my PR, where on first landing on the Files Changed page, showed all files as viewed.
![Github PR](https://user-images.githubusercontent.com/1410993/168934395-5cdfce60-e19f-4a98-91a5-c1f796d42e4d.png)

If I uncheck the file, it shows the file changes/diff and a negative viewed number and I cannot click Viewed again

**Comment by leohidalgo** [comment]
+1

**Comment by worroc** [comment]
+1
POSThttps://github.com/..../..../pull/3445/file_review [HTTP/2 404 Not Found 299ms]

**Comment by cudhayakumar** [comment]
I am too facing the same.
<img width="498" alt="image" src="https://user-images.githubusercontent.com/81548858/168979450-18d6d92f-2be9-46c1-86f6-8fea58863a4f.png">

**Comment by kayakov** [comment]
+1

**Comment by HappyRashair** [comment]
Same for me:
![image](https://user-images.githubusercontent.com/95406820/168996832-ae3ee3fb-6393-4cea-af12-c778a1dee825.png)

**Comment by AssafDamari** [comment]
+1

**Comment by motinani** [comment]
+1

**Comment by AvivNevo** [comment]
Same

**Comment by guySadger** [comment]
+1

**Comment by alex-tarasiuk-silk** [comment]
+1

**Comment by Dina-Kesselman** [comment]
+1

**Comment by JackBoosY** [comment]
Same for me.

**Comment by JackBoosY** [comment]
This is a critical github bug, I hope the maintainer can fix this as soon as possible!

**Reply by JackBoosY** [reply]
Already report this in github ticket.

**Reply by pauloalexandrecampos** [reply]
Can you please post the ticket here? 🙏

**Comment by fabien-somnier** [comment]
same for me and several people from my company.

**Reply by fabien-somnier** [reply]
I've posted a (personal) ticket for this issue too.
I don't know where to create a public ticket about GH itself (does anyone know?)

**Reply by xmo-odoo** [reply]
I've never heard of such a thing, AFAIK there's no option to do that on tickets, you possibly github employees can do that on their end?

Unless "Account or organization" "other" makes the ticket public (that seems doubtful), but you have to select that when writing the ticket.

**Reply by fabien-somnier** [reply]
it's finally resolved

**Comment by guzelgun** [comment]
These reviews will never end 😅 

<img width="353" alt="Screen Shot 2022-05-18 at 11 02 43" src="https://user-images.githubusercontent.com/11789483/169016097-5c25acb9-bf36-4203-855d-e9981fac818c.png">

**Comment by kostadin-mandulov** [comment]
+1

**Comment by jnoda** [comment]
I'm having the same problem. It seems related to the status of the `File filter`, and unchecking `Viewed files` seems to do the trick momentarily. But I haven't been able to properly "reset" it. It just reactivates whenever the page is refreshed.

![Screen Shot 2022-05-18 at 12 45 52 PM](https://user-images.githubusercontent.com/1232457/169022090-baeac991-1069-4466-8773-13f6aaf85044.png)

**Reply by vladimir-rapatsky** [reply]
@github-staff please post the public ticket link to fix this bug. And please provide ETA for fix

**Reply by vladimir-rapatsky** [reply]
I confirm that `https://github.com/.../.../pull/.../file_review` endpoint is broken

**Reply by vladimir-rapatsky** [reply]
Created a personal bug ticket: https://support.github.com/ticket/personal/0/1630104

**Comment by SABarron** [comment]
+1 for me and my colleagues

**Comment by franklinkim** [comment]
+1

**Comment by nbarbettini** [comment]
No need to reply to this thread with "+1" -- you can upvote the top message:

![image](https://user-images.githubusercontent.com/7525482/169028885-b445513b-a12d-433c-a3d4-ecc41d822d26.png)

Or click Subscribe to get email updates on the thread:

![image](https://user-images.githubusercontent.com/7525482/169028950-b87918c6-f343-4c7d-8fec-a3f116ed7969.png)

**Reply by ghost** [reply]
+1

**Comment by SVITY** [comment]
+1, the same problem 
![image](https://user-images.githubusercontent.com/2351886/169033935-72e1980e-691c-4d5a-88ba-a65ffeaa0031.png)

**Reply by nbarbettini** [reply]
Please don't reply with "+1" -- do this instead: https://github.com/github/feedback/discussions/16811#discussioncomment-2775138

**Comment by irina-capatina-uipath** [comment]
+1

**Reply by nbarbettini** [reply]
Please don't reply with "+1" -- do this instead: https://github.com/github/feedback/discussions/16811#discussioncomment-2775138

**Comment by rodrigo-araujo** [comment]
+1

**Reply by nbarbettini** [reply]
Please don't reply with "+1" -- do this instead: https://github.com/github/feedback/discussions/16811#discussioncomment-2775138

**Comment by MartinMikita** [comment]
Does not seem to help it anyway... Github thinks this is normal:

![image](https://user-images.githubusercontent.com/2283182/169041122-0267facf-f4fa-40eb-819f-d47216c3c664.png)

**Reply by xmo-odoo** [reply]
Github might have *some* monitoring noticing when the entire thing falls down, but I doubt they have e.g. bots permanently hammering their infra trying to see if everything remains consistent (and having a project which makes pretty heavy use of the API, I can assure you that github is often inconsistent in the first place at short timescales).

It's pretty common for things to start failing progressively and the status screen to remain "all green" until one of github's peeps finally notices the issue (/ gets a ticket), or the issue has spread enough that it affects most everyone.

**Comment by zhaparoff** [comment]
I've faced the same today.
Probably it is worth creating a bug report at support portal for it?
https://support.github.com/contact/bug-report

**Reply by fabien-somnier** [reply]
some of us have already done it, and an investigation is in-progress (screenshot is from today 11:30 BST)
![Screenshot 2022-05-18 at 12 15 14](https://user-images.githubusercontent.com/3116720/169054911-92c04dd3-41d5-405f-aa9c-616db11c8cde.png)

**Comment by pmat2** [comment]
![image](https://user-images.githubusercontent.com/46838259/169051283-6cd73a93-0485-425a-bec2-4632edbb32fb.png)

Negative files viewed.
After unchecking default viewed it does not allow to check it back.

**Comment by komissarovdenis** [comment]
The same issue I faced. Please, fix ASAP

**Comment by alicepaixao** [comment]
+1

**Comment by apkhandkar** [comment]
![Screenshot from 2022-05-18 08-54-34](https://user-images.githubusercontent.com/26791911/169058901-74998837-b02f-4163-ab33-27b91c9c3b8f.png)

**Reply by vladimir-rapatsky** [reply]
Please refresh. Seems to be fixed 1 minute ago

**Reply by vladimir-rapatsky** [reply]
@sam-woodridge could you please check now too?

**Reply by thetomcake** [reply]
Seems to be fixed for me

**Comment by sam-woodridge** [comment, marked as answer]
The issue seems to be resolved for me and others that I've spoken to/have posted here, as of a few hours ago.
